02 Toolbar/src/common/managers/textAreaManager.ts

```diff
@@ -0,0 +1,60 @@
+
+// Based on: http://stackoverflow.com/questions/1064089/inserting-a-text-where-cursor-is-using-javascript-jquery
+class TextAreaManager {
+  insertAtCaret(textArea: HTMLTextAreaElement, caret: string, offsetCursor: number = 0): string {
```

Comment:
Maybe we have to keep it like that _insertAtCaret_

Another option could be _insertAtCaretGetText_ but not sure if makes sense